src/test/java/seedu/todo/model/task/ValidationTaskTest.java

```diff
@@ -0,0 +1,188 @@
+        LocalDateTime startTime = LocalDateTime.of(1, 1, 1, 1, 1);
+        task.setStartTime(startTime);
+        task.validate();
+    }
+
+    @Test(expected = ValidationException.class)
+    public void testValidateTaskOnlyEndTime() throws ValidationException {
+        LocalDateTime endTime = LocalDateTime.of(1, 1, 1, 1, 1);
+        task.setEndTime(endTime);
+        task.validate();
+    }
+
+    @Test(expected = ValidationException.class)
+    public void testValidateTaskStartTimeBeforeEnd() throws ValidationException {
+        LocalDateTime startTime = LocalDateTime.of(1, 1, 1, 1, 2);
+        LocalDateTime endTime = LocalDateTime.of(1, 1, 1, 1, 1);
+
+        task.setStartTime(startTime);
+        task.setEndTime(endTime);
+
+        task.validate();
+    }
+
+    @Test
+    public void testConvertToTask() throws ValidationException {
+        LocalDateTime startTime = LocalDateTime.of(1, 1, 1, 1, 1);
+        LocalDateTime endTime = LocalDateTime.of(1, 1, 1, 1, 2);
+
+        task.setStartTime(startTime);
+        task.setEndTime(endTime);
+
+        assertAllPropertiesEqual(task, task.convertToTask());
+    }
+
+    @Test(expected = AssertionError.class)
+    public void testConvertDifferentTask() throws ValidationException {
+        Task convertedTask = task.convertToTask();
+        task.setPinned(true);
+        // task.setDescription("test");
+        assertAllPropertiesEqual(task, convertedTask);
+    }
+
+    @Test
+    public void testTaskImmutableTask() {
+        ValidationTask original = new ValidationTask("Mock Task");
+        assertAllPropertiesEqual(original, new ValidationTask(original));
+
+        original = new ValidationTask("Mock Task");
+        original.setStartTime(LocalDateTime.now());
+        original.setEndTime(LocalDateTime.now().plusHours(2));
+        assertAllPropertiesEqual(original, new Task(original));
+
+        original = new ValidationTask("Mock Task");
+        original.setDescription("A Test Description");
+        original.setLocation("Test Location");
+        assertAllPropertiesEqual(original, new ValidationTask(original));
+    }
+
+    @Test
+    public void testTitle() {
+        task.setTitle("New Title");
+        assertEquals("New Title", task.getTitle());
+    }
+
+    @Test
+    public void testDescription() {
+        assertFalse(task.getDescription().isPresent());
+
+        task.setDescription("A short description");
+        assertEquals("A short description", task.getDescription().get());
+    }
+
+    @Test
+    public void testLocation() {
+        assertFalse(task.getLocation().isPresent());
+
+        task.setLocation("Some Test Location");
+        assertEquals("Some Test Location", task.getLocation().get());
+    }
+
+    @Test
+    public void testTime() {
+        assertFalse(task.getStartTime().isPresent());
+        assertFalse(task.getEndTime().isPresent());
+
+        // TODO: Time definitely needs validation, for example task end time
```

Comment:
Please don't blindly copy and paste code without at least reading through them first